modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/fsm/Query.java

```diff
@@ -137,4 +137,10 @@ void onError(Throwable th) {
 
         resultHolder.completeExceptionally(th);
     }
+
+    CompletableFuture<Void> cancel() {
+        cancel.cancel();
+
+        return onPhaseStarted(ExecutionPhase.TERMINATED);
```

Comment:
looks like cancellation via KILL and cancellation via CancelHandle completes at different moments... We need to unify it (see last `if` in `QueryExecutor#trackQuery` for details)

Reply:
I hope I fixed it.
I introduced new methods in Query

`register(runningQueries)` -> self registration of a query in the queries registry :thinking:
and
`cancel()` -> cancels the query and `CancellationToken` now also uses this method to cancel the query.